app/src/main/java/com/example/habitsmasher/HabitDialog.java

```diff
@@ -0,0 +1,273 @@
+package com.example.habitsmasher;
+
+import android.app.DatePickerDialog;
+import android.content.Context;
+import android.util.Log;
+import android.widget.Button;
+import android.widget.DatePicker;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.view.View;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.DialogFragment;
+import com.example.habitsmasher.ui.dashboard.HabitListFragment;
+
+
+/**
+ * Abstract UI class that describes any dialog that can
+ * add or edit habits
+ * @author Jason Kim
+ */
+public abstract class HabitDialog extends DialogFragment implements DisplaysErrorMessages {
+
+    // codes for the different error messages displayed by the habit dialog
+    public static final int INCORRECT_TITLE = 1;
+    public static final int INCORRECT_REASON = 2;
+    public static final int INCORRECT_DATE = 3;
+    public static final int INCORRECT_DAYS = 4;
+
+    // tag of dialog for logging
+    protected String TAG;
+
+    // parent habit list fragment that spawned this habit dialog
+    protected HabitListFragment _habitListFragment;
+
+    // text elements of the habit dialog
+    protected EditText _habitTitleEditText;
+    protected EditText _habitReasonEditText;
+    protected TextView _habitDateTextView;
+
+    // error message text views
+    protected TextView _errorText;
```

Comment:
This text view is a text view which is initially set to contain an empty string, but will be set to contain an specified error message when prompted.